src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala

```diff
@@ -331,6 +331,32 @@ class SparkBigQueryEndToEndITSuite extends FunSuite
     initialDataValuesExist shouldBe true
   }
 
+  test("keeping filters behaviour") {
+    val newBehaviourWords = extractWords(
+      spark.read.format("bigquery")
+      .option("table","publicdata.samples.shakespeare")
+      .option("filter","length(word) = 1")
+      .option("combinePushedDownFilters","true")
+      .load())
+
+    val oldBehaviourWords = extractWords(
+      spark.read.format("bigquery")
+      .option("table","publicdata.samples.shakespeare")
+      .option("filter","length(word) = 1")
+      .option("combinePushedDownFilters","false")
+      .load())
+
+    newBehaviourWords should equal (oldBehaviourWords)
```

Comment:
Does "should equal" imply ordering here? The read API does not guarantee that reading from the same table will return results in the same order.

Reply:
No, the helper method collects the data and converts it to a set